frontend/src/containers/BatteryHealth.jsx

```diff
@@ -0,0 +1,80 @@
+import React, { useEffect, useState } from "react"
+import Box from "../components/Box"
+import { fetchDataProduct } from "../utils"
+import DataProductLink from "../components/DataProductLink"
+
+// Data definition and source to use in this example
+const DEFINITION = "DigitalProductPassport/Battery/HealthData_v0.2"
+const SOURCE = "dpp_demo:access_control_demo"
+
+// in this app we will just use the hardcoded data as parameters
+const PRODUCT = "lithium-ion"
+const ID = "660e8400-e29b"
+
+export default function BatteryHealth() {
+  const [batteryHealthData, setBatteryHealthData] = useState({
+    status: "",
+    manufacturingDate: "",
+  })
+  const [isLoading, setIsLoading] = useState(false)
+
+  // fetch data product automatically
+  useEffect(() => {
+    ;(async () => {
+      try {
+        setIsLoading(true)
+        const resp = await fetchDataProduct(
+          DEFINITION,
+          {
+            product: PRODUCT,
+            id: ID,
+          },
+          false,
+          SOURCE
+        )
+        if (resp.ok) {
+          setBatteryHealthData(resp.data)
+        } else {
+          throw new Error("Failed to fetch battery health data")
+        }
+      } finally {
+        setIsLoading(false)
+      }
+      return () => {}
+    })()
+  }, [])
+
+  if (isLoading) {
+    return (
+      <Box title={"Access control keys"}>
+        <i>Loading...</i>
+      </Box>
+    )
+  }
+
+  return (
+    <Box title={"Access control keys"}>
+      <div>
+        Battery health data for battery {PRODUCT} / {ID}:
+      </div>
+      <ul>
+        <li>
+          <b>Status:</b> {batteryHealthData.status}
+        </li>
+        <li>
+          <b>Manufacturing date:</b> {batteryHealthData.manufacturingDate}
+        </li>
+      </ul>
+      <div>
+        <p>
+          This data is available only with an application's private access control key,
+          and an API token fetched with it.
+        </p>
+        <p>
+          This is mock data generated for demo purposes only. It's fetched automatically
+          from <DataProductLink definition={DEFINITION} />
```

Comment:
You would need to also pass in the `SOURCE`to the `DataProductLink` (and do corresponding changes to it). Now it's stating that it's using the `ioxio` source whereas it is actually using the `dpp_demo:access_control_demo` source:
![image](https://github.com/user-attachments/assets/380237cb-6260-4c30-bbb2-e18ca0a28071)

Reply:
Hah, good catch, I kinda gave up on looking at that one at all since it feels broken as it is.